experiments/ppo_gridnet.py

```diff
@@ -479,39 +482,53 @@ def get_value(self, x):
-
-            ## EVALUATION LOGIC:
-            if os.path.exists("league.temp.csv"):
-                league = pd.read_csv("league.temp.csv", index_col="name")
-                if len(eval_queue) > 0:
-                    model_path = eval_queue[0]
-                    if model_path in league.index:
-                        model_global_step = int(model_path.split("/")[-1][:-3])
-                        print(f"Model global step: {model_global_step}")
-                        eval_queue = eval_queue[1:]
-                        print("charts/trueskill", league.loc[model_path]["trueskill"], model_global_step)
-                        writer.add_scalar("charts/trueskill", league.loc[model_path]["trueskill"], model_global_step)
-
-                        # Table visualization logic
-                        trueskill_data = {
-                            "name": league.loc[model_path].name,
-                            "mu": league.loc[model_path]["mu"],
-                            "sigma": league.loc[model_path]["sigma"],
-                            "trueskill": league.loc[model_path]["trueskill"],
-                        }
-                        trueskill_df = trueskill_df.append(trueskill_data, ignore_index=True)
-                        wandb.log({"trueskill": wandb.Table(dataframe=trueskill_df)})
-                        trueskill_data["type"] = "training"
-                        trueskill_data["step"] = model_global_step
-                        trueskill_step_df = trueskill_step_df.append(trueskill_data, ignore_index=True)
-                        preset_trueskill_step_df_clone = preset_trueskill_step_df.copy()
-                        preset_trueskill_step_df_clone["step"] = model_global_step
-                        trueskill_step_df = trueskill_step_df.append(preset_trueskill_step_df_clone, ignore_index=True)
-                        wandb.log({"trueskill_step": wandb.Table(dataframe=trueskill_step_df)})
+                        "--output-path",
+                        f"runs/{experiment_name}/{global_step}.csv",
+                    ],
+                    f"models/{experiment_name}/{global_step}.pt",
+                    f"runs/{experiment_name}/{global_step}.csv",
+                ]
+            ]
+            print(f"Queued models/{experiment_name}/{global_step}.pt")
+
+        while len(eval_queue) > 0:
+            # always start the max-eval-workers processes
+            for idx in range(len(eval_queue[: args.max_eval_workers])):
+                if type(eval_queue[idx][0]) == list:
+                    eval_queue[idx][0] = subprocess.Popen(eval_queue[idx][0])
+                    print(f"Evaluating {eval_queue[idx][1]}")
+
+            if eval_queue[0][0].poll() is not None:
+                # read and clean up
+                league = pd.read_csv(eval_queue[0][2], index_col="name")
+                model_path = eval_queue[0][1]
+                eval_queue = eval_queue[1:]
```

Comment:
Nitpick: it might be handy to use `collections.deque` here (it has both `.append` and `.popleft` so could be used as a proper queue)

Reply:
This is a good point! Although in the previous line we have `for idx in range(len(eval_queue[: args.max_eval_workers])):`, so we are looking at more than the top. Would deque still help?